packages/salesforcedx-vscode-replay-debugger/src/adapter/apexReplayDebug.ts

```diff
@@ -178,6 +219,75 @@ export class ApexReplayDebug extends LoggingDebugSession {
     this.sendResponse(response);
   }
 
+  protected async scopesRequest(
+    response: DebugProtocol.ScopesResponse,
+    args: DebugProtocol.ScopesArguments
+  ): Promise<void> {
+    response.success = true;
+    const frameInfo = this.logContext.getFrameHandler().get(args.frameId);
+    if (!frameInfo) {
+      response.body = { scopes: [] };
+      this.sendResponse(response);
+      return;
+    }
+    const scopes = new Array<Scope>();
+    scopes.push(
+      new Scope(
+        'Local',
+        this.logContext
+          .getScopeHandler()
+          .create(
+            new ScopeContainer(SCOPE_TYPES.LOCAL, Array.from(
+              frameInfo.locals.values()
+            ) as ApexVariable[])
+          ),
+        false
+      )
+    );
+    scopes.push(
+      new Scope(
+        'Static',
+        this.logContext
+          .getScopeHandler()
+          .create(
+            new ScopeContainer(SCOPE_TYPES.STATIC, Array.from(
+              frameInfo.statics.values()
+            ) as ApexVariable[])
+          ),
+        false
+      )
+    );
+    scopes.push(
+      new Scope(
+        'Global',
+        this.logContext
+          .getScopeHandler()
+          .create(
+            new ScopeContainer(SCOPE_TYPES.GLOBAL, Array.from(
+              frameInfo.globals.values()
+            ) as ApexVariable[])
+          ),
+        false
+      )
+    );
+    response.body = { scopes: scopes };
+    this.sendResponse(response);
+  }
+
+  protected async variablesRequest(
+    response: DebugProtocol.VariablesResponse,
+    args: DebugProtocol.VariablesArguments
+  ): Promise<void> {
+    response.success = true;
+    const scopesContainer = this.logContext
```

Comment:
Use the ID assigned in the scope creation to retrieve the ScopeContainer which holds the variables information